### PR TITLE
Adding Recipes waves

### PR DESCRIPTION
**Pull request by kbrindley**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] (Not Applicable) If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kbrindley** [comment]
This project is already a conda package with a recipe: https://github.com/lanl/waves/blob/main/recipe-github/meta.yaml and a conda package file: https://github.com/lanl/waves/releases/tag/0.6.7.

I'm struggling to find examples of conda-forge recipes that include existing Conda packages. This recipe is a first draft to re-package after ``conda-build`` unpacks the tar archive. I don't think this is the right approach and I will continue to search for a better example. In the meantime, if anyone can point me in the right direction, I think all I need is an example to start from.

Thanks!

Kyle

**Comment by dopplershift** [comment]
The usual way here would be to use the GitHub tarball for the release.

**Comment by kbrindley** [comment]
@dopplershift  As in the source code tarball ``*.tar.gz`` (as opposed to a Conda package ``*.tar.bz2``)?

I'm building and deploying on an internal channel in parallel. I was hoping to re-use the existing Conda package, but if I need to re-build from source for conda-forge, I can. I was afraid that might be the answer, so I'm working on a ``*.tar.gz`` recipe now.

Thanks!

**Comment by dopplershift** [comment]
Since it's pure Python, you're saving no effort whatsoever by trying to reuse the build. And in general, conda-forge works by having everything built by a common toolchain on shared infrastructure. The recipe here needs to be self-contained.

File: recipes/waves/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    - {{ PYTHON }} setup.py install
+    - scons man html-github
+    - {{ PYTHON }} package_documentation.py
+  ignore_run_exports:
+    - python_abi
+
+requirements:
+  build:
+    - python {{ python }}
+    - git
```
Comment (dopplershift) [inline recipes/waves/meta.yaml:27]: Is `git` needed if you're setting `SETUPTOOLS_SCM_PRETEND_VERSION`?

File: recipes/waves/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  script:
+    - {{ PYTHON }} setup.py install
+    - scons man html-github
+    - {{ PYTHON }} package_documentation.py
+  ignore_run_exports:
+    - python_abi
+
+requirements:
+  build:
+    - python {{ python }}
```
Comment (dopplershift) [inline recipes/waves/meta.yaml:26]: I don't think this is necessary? It should be enough to have the one under `host` below? 
```suggestion
```
I also don't know what that construct with the template is even trying to do...

File: recipes/waves/meta.yaml (hunk 1 above)
Comment (kbrindley) [reply recipes/waves/meta.yaml:27]: Probably not. Thanks for the catch.

File: recipes/waves/meta.yaml (hunk 2 above)
Comment (kbrindley) [reply recipes/waves/meta.yaml:26]: You may be right. I'll admit I'm a bit fuzzy on the ``host`` section. I'll re-read the current documentation and try to trim the requirements list to only those items that are strictly necessary.

The ``{{ python }}`` is a hold-over from an earlier recipe where the Python version was a build variant: https://docs.conda.io/projects/conda-build/en/stable/resources/variants.html#build-variants. It should not be necessary anymore, but I'll test locally to confirm.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**